### ISSUE TITLE
docker_compose provider cannot handle "image" statements with SHA-256-ids

### ISSUE THREAD

**Issue by mpdude** [puppetlabs/puppetlabs-docker#655]
## Describe the Bug

When `docker_compose` is pointed to a `docker-compose.yml` file that specifies the image like so:

```yml
version: '2'
services:
  my_service:
    image: melville/whale@sha256:58a102090d5e684e1956fdf0ce7bf34084e88cf91d9e423dc6c535aede19d263
    ...
```

then on _every_ Puppet run I'll see

```
Info: Checking for compose project my_project
Info: Checking for compose service my_service melville/whale@sha256:58a102090d5e684e1956fdf0ce7bf34084e88cf91d9e423dc6c535aede19d263
Info: Running compose project my_project
Notice: /Stage[main]/.../Docker_compose[my_project]/ensure: created
```

Note that I specify Docker images using their SHA265 id to ensure consistent deployments to well-known (and audited) versions; image tags may be re-assigned.

## Expected Behavior

`docker-compose ... up ...` should be run only once for every (new) image that needs to be deployed.

## Steps to Reproduce

1. Use a `docker-compose.yml` file with an `image` statement as above, including the image ID. Use the `docker_compose` resource to deploy it.
2. Run Puppet again.

## Environment
 - Version 3.11.0 of this module here

## Additional Context

My guess is the problem lies in the `exists?` method in `lib/puppet/provider/docker_compose/ruby.rb`:

The list of currently running containers for the compose project is parsed from `docker ps` output, using `{{.Label "com.docker.compose.service"}}-{{.Image}}` as the format string. This does not include the image hashes/ids.

Next, this list is compared to information parsed from `docker-compose.yml`. This time, the `image` contains the `@sha265:...` suffix.

Since both strings don't match, the provider assumes the project needs to be rebuilt.

**Comment by mpdude** [comment]
Yes, it is still important and I also suggested a fix in #814.

**Comment by towo** [comment]
This is still relevant, for what it's worth.
